src/Persistence_matrix/include/gudhi/Persistence_matrix/overlay_ididx_to_matidx.h

```diff
@@ -756,7 +756,7 @@ inline void Id_to_index_overlay<Matrix_type, Master_matrix_type>::insert_boundar
     GUDHI_CHECK(idToIndex_->find(faceIndex) == idToIndex_->end(),
                 std::invalid_argument("Id_to_index_overlay::insert_boundary - Index for simplex already chosen!"));
   } else {
-    GUDHI_CHECK((idToIndex_->size() <= faceIndex || idToIndex_[faceIndex] == static_cast<index>(-1)),
+    GUDHI_CHECK((idToIndex_->size() <= faceIndex || idToIndex_->operator[](faceIndex) == static_cast<index>(-1)),
```

Comment:
I find both not really readable. So I have no preference. Do you want me to change it?

Reply:
And `_id_to_index(faceIndex)` is not good either?

Reply:
No, for readability, this would probably be the best option. I just didn't thought that `idToIndex_->operator[](faceIndex)` was so unreadable that it was necessary. But no problem to change that.

Reply:
It isn't unreadable, but it looks strange to have a lot of ugly calls to `operator[]` when the whole point of operators is their nice short syntax.

Reply:
I agree, it is a bit of a pity that those operators doesn't work similarly with pointers.